pkg/controller/direct/certificatemanager/dnsauthorization_controller.go

```diff
@@ -0,0 +1,334 @@
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+
+	certificatemanagerpb "cloud.google.com/go/certificatemanager/apiv1/certificatemanagerpb"
+	kccpredicate "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/predicate"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ctrlName      = "certificatemanager-controller"
+	serviceDomain = "//certificatemanager.googleapis.com"
+)
+
+func init() {
+	rg := &DNSAuthorizationReconcileGate{}
+	registry.RegisterModelWithReconcileGate(krm.CertificateManagerDNSAuthorizationGVK, NewModel, rg)
+}
+
+type DNSAuthorizationReconcileGate struct {
+	optIn kccpredicate.OptInToDirectReconciliation
+}
+
+var _ kccpredicate.ReconcileGate = &DNSAuthorizationReconcileGate{}
+
+func (r *DNSAuthorizationReconcileGate) ShouldReconcile(o *unstructured.Unstructured) bool {
+	return r.optIn.ShouldReconcile(o)
+}
+
+func NewModel(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
+	return &model{config: *config}, nil
+}
+
+var _ directbase.Model = &model{}
+
+type model struct {
+	config config.ControllerConfig
+}
+
+func (m *model) client(ctx context.Context) (*gcp.Client, error) {
+	var opts []option.ClientOption
+	opts, err := m.config.RESTClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	gcpClient, err := gcp.NewRESTClient(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("building DnsAuthorization client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *unstructured.Unstructured) (directbase.Adapter, error) {
+	obj := &krm.CertificateManagerDNSAuthorization{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		return nil, fmt.Errorf("error converting to %T: %w", obj, err)
+	}
+
+	// Get ResourceID
+	resourceID := direct.ValueOf(obj.Spec.ResourceID)
+	if resourceID == "" {
+		resourceID = obj.GetName()
+	}
+	if resourceID == "" {
+		return nil, fmt.Errorf("cannot resolve resource ID")
+	}
+
+	projectRef, err := refs.ResolveProject(ctx, reader, obj, &obj.Spec.ProjectRef)
+	if err != nil {
+		return nil, err
+	}
+	projectID := projectRef.ProjectID
+	if projectID == "" {
+		return nil, fmt.Errorf("cannot resolve project")
+	}
+
+	// Get location
+	// TODO: Add location when field is added
+	// location := obj.Spec.Location
+	location := "global"
+
+	var id *CertificateManagerDNSAuthorizationIdentity
+	// TODO: Add ExternalRef when field is added
+	// externalRef := direct.ValueOf(obj.Status.ExternalRef)
+	externalRef := ""
```

Comment:
Is this intended?

Reply:
yes, the field is currently not implemented. I just leave this code here when we enable the field (see TODO on top).